### PR TITLE
Badges

### PR DESCRIPTION
**Pull request by Hackante**

### CODE REVIEW

File: commands/staff/player.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+        if(!target) return message.channel.send({content:`Invalid player! Use\n\`${this.usage}\``})
+        if(!columns.includes(column)) return message.channel.send({content:`Invalid column! Available columns are ${columns.map((column) => `\`${column}\``).join(" ")}`})
+        if(!operators.includes(operator)) return message.channel.send({content:`Invalid operator! Available operators are ${operators.map((operator) => `\`${operator}\``).join(" ")}`})
+        if(!value) return message.channel.send({content:`Invalid usage! Use\n\`${this.usage}\``})
+        if(options !== "none" && ["-f", "force", "--force"].includes(options)) force = true
+
+        let playerData = await players.findOne({user: target.id})
+
+        if(column === "coins" || column === "roses" || column === "gems" || column === "xp" || column === "rose" || column === "bouquet" || column === "lootbox") {
```
Comment (thewilloftheshadow) [inline commands/staff/player.js:37]:
```
if(["coins", "roses", etc].includes(column) {

File: events/guildMemberAdd.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        client.invites.every(async (invite) => {
+            let inv = await sim.invites.resolve(invite.code)
+            if(inv.uses !== invite.uses) {
+                await players.updateOne({"badges.invite.code": invite.code}, {$inc:{"badges.invite.members": 1}})
+                let guy = await players.findOne({"badges.invite.code": invite.code})
+                if(guy.badges.invite.unlocked === true) return
+                if(guy.badges.invite.members = 15) {
+                    await players.updateOne({"badges.invite.code": invite.code}, {$set:{"badges.invite.unlocked": true}})
+                    let guyUser = client.users.resolve(guy.user)
+                    guyUser.send({content:`Congratulations! You just unlocked the invite badge with the code \`${invite.code}\``})
```
Comment (thewilloftheshadow) [inline events/guildMemberAdd.js:17]: Use i10n for this

File: events/slashCmds.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
+const db = require("quick.db")
+const config = require("../config")
+const { Collection, Util } = require("discord.js")
+const cooldowns = new Collection()
+
+module.exports = (client) => {
+    client.on("interactionCreate", async (interaction) => {
+        if (!interaction.isCommand) return
```
Comment (thewilloftheshadow) [inline events/slashCmds.js:8]: isCommand is a function not a property

File: commands/bot/badges.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+function capitalizeFirstLetter(string) {
+    return string.charAt(0).toUpperCase() + string.slice(1)
+}
+
+module.exports = {
+    name: "badges",
+    description: "Displays all your badges.",
+    usage: `${process.env.PREFIX}badges [user]`,
+    run: (message, args, client) => {
+        let guy = message.mentions.members.first() || message.guild.members.cache.get(args[0]) || message.guild.members.cache.find((x) => x.user.username.toLowerCase() === args.slice(0).join(" ") || x.user.username === args[0] || x.user.tag === args[0])
```
Comment (thewilloftheshadow) [inline commands/bot/badges.js:13]: the fn from config has a getUser function you can use instead of this

**Review by thewilloftheshadow — CHANGES_REQUESTED** [review]
Looks great so far, this is just what I saw with a glance through it before I drive home

**Comment by thewilloftheshadow** [comment]
Couple things to think about:

- [ ] Should the design of badges be emojis like the badges in the app? Or just text on the profile. Or both?
- [ ] Should we also add badges for things like staff, beta tester, playing certain number of games
- [ ] Can we switch to only using guild slashCommands, since the bot is only for specific guilds and they can update instantly instead of the bot-wide 1 hour cooldown for regular ones

**Comment by Hackante** [comment]
> * [ ]  Should the design of badges be emojis like the badges in the app? Or just text on the profile. Or both?
I would say yes but I thought we can add this later after the halloween event as these badges should be done quickly

> * [ ]  Should we also add badges for things like staff, beta tester, playing certain number of games
Sure! I think we should ask other staffs to help us making a badge list

> * [ ]  Can we switch to only using guild slashCommands, since the bot is only for specific guilds and they can update instantly instead of the bot-wide 1 hour cooldown for regular ones
This is actually the system behind the file reloadslash.js (cmd.server.forEach(...client.application.commands.create(cmd.command, id)))

> Also don't forget to check the error testing workflow for small checks that failed
Sure ^^

File: events/guildMemberAdd.js (hunk 2 above)
Comment (Hackante) [reply events/guildMemberAdd.js:17]: So I should replace every string with an i10n string? or just this one?

And this also implements creating prs on i10n right?

Comment (thewilloftheshadow) [reply events/guildMemberAdd.js:17]: Just the strings that are public, not the error stuff for staff.
PR it to the bottom of the en.json file for the English translation

**Comment by thewilloftheshadow** [comment]
> > * [ ]   Can we switch to only using guild slashCommands, since the bot is only for specific guilds and they can update instantly instead of the bot-wide 1 hour cooldown for regular ones
> 
> This is actually the system behind the file reloadslash.js (cmd.server.forEach(...client.application.commands.create(cmd.command, id)))

ah ok, i usually use message.guild.commands, thats why I didn't see that

<hr>

Looks good :)

File: events/guildMemberAdd.js [hunk 5]
```diff
@@ -27,6 +27,22 @@ module.exports = (client) => {
                     guyUser.send({ content: member.i10n("inviteBadgeUnlocked", { code: invite.code }) })
                 }
             }
-        })
+        })*/
+
+        // Mainly copied from Wolvesville Utopium Bot
```
Comment (thewilloftheshadow) [inline events/guildMemberAdd.js:32]: And i copied that code from stackoverflow into wolvesville utopium xD

**Comment by thewilloftheshadow** [comment]
Can we just drop the invite badge portion of this so we can merge the rest of the framework for October events?

**Comment by Hackante** [comment]
Yes we can. I'll just comment out the code for the invite badge

**Review by thewilloftheshadow — APPROVED** [review]

**Pull request merged**